service/src/main/java/org/apache/griffin/core/job/JobServiceImpl.java

```diff
@@ -606,6 +610,10 @@ private void setJobInstanceIdAndUri(JobInstanceBean instance, HashMap<String
             instance.setAppUri(appId == null ? null : env
                 .getProperty("yarn.uri") + "/cluster/app/" + appId);
             instanceRepo.save(instance);
+            // If Livy returns to success or dead, task execution completes one,TaskNum--
+            if ("SUCCESS".equals(state) || "DEAD".equals(state)) {
```

Comment:
It feels more correct to compare it to enum values, than strings (or at least, toString of enum values).